src/styles/calendar.scss

```diff
@@ -0,0 +1,43 @@
+.calendar {
+  @include flex (flex, false, false, row);
+  flex-wrap: wrap;
+  width: $calendar-width;
+  gap: $gap;
+  padding: $calendar-padding;
+
+  &__day {
+    @for $index from 1 through 31 {
+      &:nth-child(#{$index})::before {
+        content: "#{$index}";
+        @include flex (flex, center, center, row);
+        width: $square-size;
+        height: $square-size;
+        background-color: $square-color;
+        border: 1px solid #000;
+      }
+
+      &:hover::before {
+        cursor: pointer;
+        background-color: $pink-color;
+        transition: 0.5s;
```

Comment:
Make sure to add transition style under general selector, not the one with :hover - this way transition will work smoothly both ways.

Also it's better to specify properties you change on hover to improve optimization